test/jdk/javax/swing/JTable/bug4460382.java

```diff
@@ -0,0 +1,124 @@
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.MenuKeyEvent;
+import javax.swing.table.DefaultTableModel;
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableModel;
+
+/*
+ * @test
+ * @bug 4460382
+ * @summary Tests that table editor works
+ * @key headful
+ * @run main bug4460382
+ */
+
+public class bug4460382 {
+    static JFrame frame;
+    static Table table;
+    static volatile Point tableLoc;
+    static volatile Point p;
+    static int row1 = -1;
+    static int row2 = -1;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(250);
+            SwingUtilities.invokeAndWait(() -> createTestUI());
+            robot.waitForIdle();
+
+            SwingUtilities.invokeAndWait(() -> {
+                tableLoc = table.getLocationOnScreen();
+                p = table.getCellRect(0, 0, true).getLocation();
+            });
+
+            robot.mouseMove(tableLoc.x + p.x + 10, tableLoc.y + p.y + 10);
+            robot.waitForIdle();
+
+            robot.mousePress(MouseEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(MouseEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+
+            robot.keyPress(MenuKeyEvent.VK_A);
+            robot.keyRelease(MenuKeyEvent.VK_A);
+            robot.keyPress(MenuKeyEvent.VK_ENTER);
+            robot.keyRelease(MenuKeyEvent.VK_ENTER);
+            robot.waitForIdle();
+
+            if (row1 != row2) {
+                throw new RuntimeException("Failed 4460382: editingRow is " + row2);
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    public static void createTestUI() {
+        frame = new JFrame("bug4460382");
+        table = new Table();
+        frame.add(new JScrollPane(table));
+        frame.setLocationRelativeTo(null);
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+    static class Table extends JTable {
+        public Table() {
+            String[][] data = {{"Click and Edit here"},
+                    {"Click and Edit here"},
+                    {"Click and Edit here"}};
+            String[] columnNames = {"Editable Column"};
+
+            TableModel model = new DefaultTableModel(data, columnNames) {
+                public boolean isCellEditable(int rowIndex, int columnIndex) {
+                    return true;
+                }
+            };
+
+            setModel(model);
+        }
+
+        public void editingStopped(ChangeEvent e) {
+            TableCellEditor editor = getCellEditor();
+            Object value = editor.getCellEditorValue();
+            row1 = editingRow;
+            setValueAt(value, editingRow, editingColumn);
+            row2 = editingRow;
```

Comment:
Can you tell me what are we testing here? We extending JTable and we are checking that during the cell content replacement some random internal protected variable does not change. What does it tells us? Why is it a criteria of test passing or not?

Reply:
i think this test should probably be removed, since the manual test was also like this in that it didn't really test anything.